scripts/deploy.sh

```diff
@@ -37,24 +45,32 @@ which openstack-ansible || ./scripts/bootstrap-ansible.sh
 ./scripts/pw-token-gen.py --file /etc/openstack_deploy/user_extras_secrets.yml
 
 # begin the openstack installation
-cd "${OSAD_DIR}"/playbooks/
+if [[ "${DEPLOY_OSAD}" == "yes" ]]; then
+  cd ${OSAD_DIR}/playbooks/
 
-# setup the hosts and build the basic containers
-install_bits setup-hosts.yml
+  # ensure that the ELK containers aren't created if they're not
+  # going to be used
+  if [[ "${DEPLOY_ELK}" != "yes" ]]; then
+    rm -f /etc/openstack_deploy/env.d/{elasticsearch,logstash,kibana}.yml
+  fi
 
-# setup the haproxy load balancer
-if [[ "${DEPLOY_HAPROXY}" == "yes" ]]; then
-  install_bits haproxy-install.yml
-fi
+  # setup the hosts and build the basic containers
+  install_bits setup-hosts.yml
+
+  # setup the haproxy load balancer
+  if [[ "${DEPLOY_HAPROXY}" == "yes" ]]; then
+    install_bits haproxy-install.yml
```

Comment:
This doesn't change anything either - but the change seems unnecessary. Perhaps @odyssey4me has a reason for changing the ordering of haproxy?

In a chronological sense, you only need haproxy setup before running infra/os plays so that the db/keystone etc will work - so it does make sense to have it after host setup in some ways - there is no need for haproxy before the host setup.